x-pack/libbeat/tests/integration/oteltest_framework.go

```diff
@@ -0,0 +1,178 @@
+)
+
+var schemeMap = map[string]string{
+	"filebeat": "fb",
+}
+
+type TestCollector struct {
+	t            *testing.T
+	tempDir      string
+	otelcol      *otelcol.Collector
+	wg           sync.WaitGroup
+	configFile   string
+	beatname     string
+	observedLogs *observer.ObservedLogs
+}
+
+// NewTestCollector configures and returns an instance of otel collector intended for testing only
+func NewTestCollector(t *testing.T, beatname string, config string) (*TestCollector, error) {
+	// create a temp dir at beat/build/integration-tests
+	tempDir := integration.CreateTempDir(t)
+
+	// create a configfile
+	configFile := filepath.Join(tempDir, beatname+".yml")
+
+	testCol := &TestCollector{
+		t:          t,
+		tempDir:    tempDir,
+		wg:         sync.WaitGroup{},
+		configFile: configFile,
+		beatname:   beatname,
+	}
+
+	// write configuration to a file
+	err := testCol.reloadConfig(config)
+	return testCol, err
+}
+
+// NewTestStartCollector configures and starts an otel collector intended for testing only
+func NewTestStartCollector(t *testing.T, beatname string, config string) (*TestCollector, error) {
+	otelcol, err := NewTestCollector(t, beatname, config)
+	if err != nil {
+		return nil, err
+	}
+	err = otelcol.Run()
+	if err != nil {
+		return nil, err
+	}
+
+	return otelcol, err
+}
+
+// reloadConfig reloads configuration with which collector should be started.
+// Note: A running collector will not pick the new config until it is stopped and started
+func (c *TestCollector) reloadConfig(config string) error {
+	// write configuration to a file
+	if err := os.WriteFile(c.configFile, []byte(config), 0o644); err != nil {
+		return fmt.Errorf("cannot create config file '%s': %s", c.configFile, err)
+	}
+	// adds scheme name as prefix to the configfile
+	beatCfg := schemeMap[c.beatname] + ":" + c.configFile
+	// get collector settings
+	set, observedLogs := getCollectorSettings(beatCfg)
+
+	c.observedLogs = observedLogs
+	// get new collector instance
+	otelcol, err := otelcol.NewCollector(set)
+	c.otelcol = otelcol
+	return err
+}
+
+// ReloadCollectorWithConfig reloads the collector with given config
+func (c *TestCollector) ReloadCollectorWithConfig(config string) error {
+	c.t.Helper()
+	// shutdown collector if it is running
+	c.Shutdown()
+	err := c.reloadConfig(config)
+	if err != nil {
+		return err
+	}
+	return c.Run()
+}
+
+func (c *TestCollector) GetTempDir() string {
+	return c.tempDir
+}
+
+// Run starts the otel collector
+func (c *TestCollector) Run() error {
+	c.t.Helper()
+
+	wg := sync.WaitGroup{}
+	var err error
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		err = c.otelcol.Run(c.t.Context())
+	}()
+
+	c.t.Cleanup(func() {
+		c.Shutdown()
+		return
+	})
+
+	return err
+}
+
+func (c *TestCollector) Shutdown() {
+	c.otelcol.Shutdown()
+}
+
+// IsCollectorHealthy returns true if collector is healthy
+func (c *TestCollector) IsCollectorHealthy(config string) bool {
+	// TODO
+	return true
+}
+
```

Comment:
We can have a function that that returns true when the collector is ready, something like this:

```go
func (c *TestCollector) IsReady() bool {
	return c.observedLogs.FilterMessage("Everything is ready. Begin running and processing data.").Len() > 0
}

Then we can use it like this in the test:

```go
	require.Eventually(t, func() bool {
		return col.IsReady()
	}, 10*time.Second, 100*time.Millisecond, "collector is not ready")
```